### PR TITLE
doc: emphasize building examples on master branch should follow COMPILING

### PR DESCRIPTION
**Pull request by dapengzhang0**

### CODE REVIEW

**Comment by dapengzhang0** [comment]
#5115 #5083 etc.

File: examples/README.md [hunk 1]
```diff
@@ -9,8 +9,11 @@ You may want to read through the
 [Quick Start Guide](https://grpc.io/docs/quickstart/java.html)
 before trying out the examples.
 
-To build the examples, run in this directory:
+To build the examples,
 
+1. **(Only need this step for non-released versions, e.g. master HEAD) [Install gRPC Java 1.18.0-SNAPSHOT library locally](../COMPILING.dm).** <!--- CURRENT_GRPC_VERSION -->
```
Comment (sanjaypujare) [inline examples/README.md]: 2 points:

- can we avoid "hardcoding" 1.18.0-SNAPSHOT in this README file? People may not remember changing this when the HEAD points to something newer.

- put the text in parentheses after the main sentence i.e.

 [Install gRPC Java 1.18.0-SNAPSHOT library locally](../COMPILING.dm) (Only need this step for non-released versions, e.g. master HEAD)

**Comment by dapengzhang0** [comment]
PTLA

**Comment by dapengzhang0** [comment]
Added clarification in COMPILING.md as well, I think that's where @sanjaypujare got confused from.

**Comment by dapengzhang0** [comment]
@ejona86 PTALA

**Pull request merged**